inc/ticket.class.php

```diff
@@ -307,6 +307,11 @@ static public function displayVoucherInTicketProcessingForm($params) {
          return;
       }
 
+      // Item might be another CommonITILObject, we only want Ticket
+      if (!$params['options']['item'] instanceof Ticket) {
```

Comment:
This will break `TicketTask` case.
You should just replace exception throwing by a `return;` on line 322.

Reply:
This should not break TicketTask as this condition is on the parent type :
`$item` can be an ITILSolution, TicketTask or ITILFollowup.
$params['options']['item']` is the parent of `$item` so it can be Ticket, Problem or Change.

Am I missing something ?

Reply:
If I remember right, for `TicketTask`, `$params['options']['item']` is not defined, so it cannot be an instance of `Ticket`.
However, tests mades 2 lines above were made to find the parent item in the more appropriate field, so result can be used to check if the found Itil object is a Ticket or not.

Reply:
I see, I've replaced the exception with a return as you suggested and simplified a bit the conditions.

Reply:
I've reverted the condition (just in case, as we discussed).